app/assistify/migrations/server/startup/migrations.js

```diff
@@ -24,3 +26,50 @@ Meteor.startup(() => {
 	});
 
 });
+
+
+/*
+ Migrate from Threads to Discussions.
+ - Messages with type 'thread-welcome' will be removed.
+ - Messages with type 'create-thread' will be migrated to 'disscussion-created'.
+ - Author Information will be encapsulated inside attachments of message type 'disscussion-created'.
+ */
+
+Meteor.startup(() => {
+	console.log('Migrating from threads to Discussions');
+	Messages.remove({ t: 'thread-welcome' });
+	Messages.find({ t: 'create-thread' }).forEach((msg) => {
+		const update = {
+			$set: {
+				t: 'discussion-created',
+			},
+			$unset: {
+				mentions: 1,
+				channels: 1,
+				urls: 1,
+			},
+		};
+		const room = Rooms.findOne({ _id: msg.channels[0]._id },
+			{
+				fields: {
+					_id: 1,
+					fname: 1,
+					lm: 1,
+					msgs: 1,
+				},
+			});
+		if (room) {
+			update.$set = {
+				msg: room.fname,
+				drid: room._id,
+				dlm: room.lm,
+				dcount: room.msgs,
+			};
+		}
+		if (msg.attachments && msg.attachments.length) {
```

Comment:
@vickyokrm what about messages not having an attachment?
Shouldn't we create a new one in this case?

Reply:
What happens if the origin message actually has a real attachment?

Reply:
@mrsimpson the system messages with type 'create-thread' will  only have attachment always as per our design. Couldn't think of any scenario upfront where there wont be no attachment.
Secondly, the system messages will not have any real attachments.
This convinced me to use array indexing.

Reply:
👌